### PR TITLE
Allows for My to support Supported Brands

### PR DESCRIPTION
**Pull request by zsarnett**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/integration.ts [hunk 1]
```diff
@@ -62,3 +70,82 @@ export const fetchIntegrationManifest = (
 
 export const fetchIntegrationSetups = (hass: HomeAssistant) =>
   hass.callWS<IntegrationSetup[]>({ type: "integration/setup_info" });
+
+export const protocolIntegrationPicked = async (
```
Comment (balloob) [inline src/data/integration.ts:74]: This shouldn't be in a file in the data folder

Comment (balloob) [reply src/data/integration.ts:74]: This folder is only for data fetching and processing

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 2]
```diff
@@ -106,6 +109,85 @@ const groupByIntegration = (
   return result;
 };
 
+export const protocolIntegrationPicked = async (
```
Comment (balloob) [inline src/panels/config/integrations/ha-config-integrations.ts:112]: just move it to it's own file

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 3]
```diff
@@ -677,49 +759,92 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
...
+    const supportedBrands = await getSupportedBrands(this.hass);
+    const supportedBrandsIntegrations = {};
+
+    for (const [d, domainBrands] of Object.entries(supportedBrands)) {
+      for (const [slug, name] of Object.entries(domainBrands)) {
+        supportedBrandsIntegrations[slug] = {
+          name,
+          supported_flows: [d],
+        };
       }
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integrations.ts:796]: Didn't we create a helper for this? If not we probably should

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 4]
```diff
@@ -677,49 +759,92 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
...
+        }))
+      ) {
         return;
       }
-      const helpers = await getConfigFlowHandlers(this.hass, "helper");
-      if (helpers.includes(domain)) {
-        navigate(`/config/helpers/add?domain=${domain}`, {
-          replace: true,
-        });
-        return;
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integrations.ts:694]: This is missing now?

File: src/panels/config/integrations/ha-config-integrations.ts (hunk 3 above)
Comment (zsarnett) [reply src/panels/config/integrations/ha-config-integrations.ts]: Its not the same as the other need. But I did make one for this

**Review by bramkragten — APPROVED** [review]

**Pull request merged**